### ISSUE TITLE
HW Loop produces an ICE during building

### ISSUE THREAD

**Issue by Slattz** [openhwgroup/corev-gcc#101]
Since the roll-forward a few weeks ago, building the latest commit is producing an ICE when XCVhwlp is enabled.
```
/home/user/git/openhw/install/riscv32-corev-elf/bin/ranlib libgcov.a
../../../../../../gcc/libgcc/unwind-dw2-fde.c: In function 'fde_radixsort':
../../../../../../gcc/libgcc/unwind-dw2-fde.c:674:1: error: insn does not satisfy its constraints:
  674 | }
      | ^
(insn 430 429 431 19 (set (reg:SI 70 lpstart0)
        (reg:SI 14 a4 [266])) 29242 {*movsi_internal}
     (expr_list:REG_DEAD (reg:SI 14 a4 [266])
        (nil)))
during RTL pass: cprop_hardreg
../../../../../../gcc/libgcc/unwind-dw2-fde.c:674:1: internal compiler error: in extract_constrain_insn, at recog.cc:2713
0xa92ed9 _fatal_insn(char const*, rtx_def const*, char const*, int, char const*)
        ../../../gcc/gcc/rtl-error.cc:108
```
`movsi_internal` is redefined in [gcc/config/riscv/corev.md](https://github.com/openhwgroup/corev-gcc/blob/development/gcc/config/riscv/corev.md). The [original commit](https://github.com/openhwgroup/corev-gcc/commit/7e1226591bd8eeacb447b613ea1a34584ae3ecef#diff-98b71abedd6e59eb80cb43e75ebb507ffa8bf540ee88aa099c6c7fe7cf90ae3eL2191-R2202) specified `move_dest_operand` rather than the current `nonimmediate_nonpostinc`, but changing this to how it was in the original patch doesn't fully fix the issue.

# To reproduce:
Compile the current CORE-V GCC development branch (currently based on commit a6c793a6f00854524c0a9aafef993ba065086d04) with `xcvhwlp` enabled. (e.g. `rv32im_zicsr_xcvhwlp_xcvmem_xcvmac_xcvbi_xcvalu_xcvsimd_xcvbitmanip-ilp32`)

**Comment by MaryBennett** [comment]
Does adding ```xcvl0c``` and ```xcvl1c``` as well as ```move_dest_operand``` to the riscv.md ```movsi_internal``` and the corev.md non-XCVMEM ```movsi_internal``` fix it?

**Comment by Slattz** [comment]
Fixed by PR #103
